eclair-core/src/test/scala/fr/acinq/eclair/io/PeerSpec.scala

```diff
@@ -389,6 +389,21 @@ class PeerSpec extends TestkitBaseClass with StateTestsHelperMethods {
     assert(transport.expectMsgType[Pong].data.size === ping.pongLength)
   }
 
+  test("send a ping if no message received for 30s") { f =>
+    import f._
+    connect(remoteNodeId, authenticator, watcher, router, relayer, connection, transport, peer)
+    // we make the transport send a message, this will delay the sending of a ping
+    val dummy = updates.head
+    for (_ <- 1 to 5) {
```

Comment:
IIUC the reason you're making many messages at an interval < 30 seconds is to verify that those messages correctly delay the ping sent by the peer.
I think there should be a better comment to explain that, my first reaction was "why do you do that loop?".

Reply:
Right, the intent is definitely not obvious.